k-distribution/tests/regression-new/issue-2287-simpl-rules-in-kprovex/a1-spec.k

```diff
@@ -0,0 +1,20 @@
+requires "test.k"
+
+
+module VERIFICATION
+  import TEST
+
+  rule (X +Int Y) +Int Z => Z +Int (X +Int Y) [simplification, symbolic(X), concrete(Y, Z)]
+  rule (X +Int Y) +Int Z => Y +Int (X +Int Z) [simplification, concrete(X), symbolic(Y), concrete(Z)]
+  rule X +Int (Y +Int Z) => Z +Int (X +Int Y) [simplification, concrete(X, Y), symbolic(Z)]
+  rule X +Int (Y +Int Z) => Y +Int (X +Int Z) [simplification, concrete(X), symbolic(Y), concrete(Z)]
+  rule final(n +Int _M:Int) => 0 [simplification, concrete(_M)]
+endmodule
+
```

Comment:
We should have associated error checks too, specifically for:

- non-`simplification` rules being added should be an error? (not sure if the haskell backend allows normal axiomatic rules here, but I think we should restrict this behavior anyway, since introducing new semantic rules changes the transition system, which is usually probably not what you want)
- Adding `syntax` ... declarations should be an error.
- Adding a `rule <k> ... </k> ` semantic rule (with a configuration) should be an error.

Reply:
We should also have some `simplification` rules with side-conditions in this test. Maybe just have your last rule on line `11` have a side condition?